packages/protocol/contracts/governance/Election.sol

```diff
@@ -845,4 +883,40 @@ contract Election is
     }
     return res;
   }
+
+  struct SlashVotesInfo {
+    address[] groups;
+    uint256 remainingValue;
+  }
+
+  // Iterates in reverse order over all groups voted for by `account`,
+  // reducing the total amount of voting gold by `value` in total.
+  // For each group, if there `value` gold has left to be slashed, reduces
+  // the number of pending and active votes for that group by as much as
+  // possible without exceeding the total value that needs to be slashed.
+  function slashVotes(
+    address account,
+    uint256 value,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external nonReentrant returns (bool) {
```

Comment:
I guess here should be the check that only whitelisted contracts can call.

Reply:
I think ideally we want it to be only callable from a fn within locked gold contract, where the fn within locked gold will only allow whitelisted contracts. 
I can't get this to work in the tests currently (described in the PR description) so I'm looking for solutions.